packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/Palette.tsx

```diff
@@ -240,17 +247,7 @@ export const Palette = ({ x, y, diagramElementId, onDirectEditClick, isDiagramEl
         }
       }
     }
-  }, [paletteLoading, paletteData, paletteError, setPalette, isSingleClickOnDiagramElementTool, isDiagramDescription]);
-
-  useEffect(() => {
-    getPalette({
-      variables: {
-        editingContextId: editingContextId,
-        diagramId,
-        diagramElementId,
-      },
-    });
-  }, [editingContextId, diagramId, getPalette]);
+  }, [paletteLoading, paletteData, paletteError]);
```

Comment:
Errors should be properly handled and the palette does not need to be in the state either, there's one more rendering step to remove here. I'll update that myself.